compiler/code-gen/vertex-compiler.cpp

```diff
@@ -1991,25 +1991,36 @@ void compile_callback_of_builtin(VertexAdaptor<op_callback_of_builtin> root, Cod
    * Will be transformed to:
    *   array_map([captured1 = $extern] (auto &&... args) {
    *       return lambda$xxx(captured1, std::forward<decltype(args)>(args)...);
-   *   }), const_array);
+   *   }, const_array);
+   * or in K2 mode:
+   *   co_await array_map([captured1 = $extern] (auto &&... args) [[-> task_t<lambda's return type>]]? {
+   *       [[return lambda$xxx(captured1, std::forward<decltype(args)>(args)...);
+   *       | co_return(co_await lambda$xxx(captured1, std::forward<decltype(args)>(args)...))]]
+   *   }, const_array);
    * Where the body calls a lambda function:
    *    int lambda$xxx(int $extern, int $x) { return $xx + $extern; }
    * Captured vars are not always op_var. For example, [captured1 = check_not_false($extern).val()] after smart casts.
    */
   W << LockComments{};
 
   int idx = 0;
-  W << "[" << JoinValues(*root, ", ", join_mode::one_line, [&idx](CodeGenerator &W, VertexPtr cpp_captured) {
-    W << "captured" << (++idx) << " = " << cpp_captured;
-  }) << "]";
+  W << "["
+    << JoinValues(*root, ", ", join_mode::one_line, [&idx](CodeGenerator &W, VertexPtr cpp_captured) { W << "captured" << (++idx) << " = " << cpp_captured; })
+    << "]";
+
```

Comment:
Let's leave it as is, to have a valid blame

Reply:
Yep, fixed